libnmstate/nm/applier.py

```diff
@@ -79,28 +81,33 @@ def prepare_edited_ifaces_configuration(ifaces_desired_state):
 
 
 def set_ifaces_admin_state(ifaces_desired_state):
+    devs_actions = {}
     for iface_desired_state in ifaces_desired_state:
+        devs = _get_devs_from_iface_state(iface_desired_state)
         if iface_desired_state['state'] == 'up':
-            _set_dev_state(iface_desired_state, device.activate)
+            devs_actions.update({dev: device.activate for dev in devs})
         elif iface_desired_state['state'] == 'down':
-            _set_dev_state(iface_desired_state, device.delete)
+            devs_actions.update({dev: device.delete for dev in devs})
         elif iface_desired_state['state'] == 'absent':
-            _set_dev_state(iface_desired_state, device.delete)
+            devs_actions.update({dev: device.delete for dev in devs})
         else:
             raise UnsupportedIfaceStateError(iface_desired_state)
 
+    for dev, action in six.viewitems(devs_actions):
+        action(dev)
+
 
-def _set_dev_state(iface_state, func_action):
+def _get_devs_from_iface_state(iface_state):
     nmdev = device.get_device_by_name(iface_state['name'])
+    devs = []
     if nmdev:
-        devs = [nmdev]
+        devs += [nmdev]
```

Comment:
Why do you use `+=` here? AFAICS `devs` will always be empty so it can just be assigned to the new list.

Reply:
Just to express it is already defined. After it was defined at the top, one may assume it will not get overwritten afterwards. Feels safer for future changes.